bftengine/src/preprocessor/messages/ClientPreProcessRequestMsg.cpp

```diff
@@ -28,12 +28,12 @@ ClientPreProcessRequestMsg::ClientPreProcessRequestMsg(NodeIdType sender,
   msgBody_->msgType = MsgCode::ClientPreProcessRequest;
 }
 
-unique_ptr<MessageBase> ClientPreProcessRequestMsg::convertToClientRequestMsg(bool resetPreProcessFlag) {
+unique_ptr<MessageBase> ClientPreProcessRequestMsg::convertToClientRequestMsg(bool resetPreProcessFlag, bool empty) {
   if (resetPreProcessFlag) msgBody()->flags &= ~(1 << 1);
   unique_ptr<MessageBase> clientRequestMsg = make_unique<ClientRequestMsg>(clientProxyId(),
                                                                            flags(),
                                                                            requestSeqNum(),
-                                                                           requestLength(),
+                                                                           empty ? 0 : requestLength(),
                                                                            requestBuf(),
```

Comment:
I guess we don't need to pass and then copy request buffer either.

Reply:
if the size is 0, then nothing will be copied